src/lib.rs

```diff
@@ -61,9 +62,15 @@ pub fn decode(type_variant: String, xdr_base64: String) -> Result<String, String
     let decoded_max_len = xdr_base64.len() / 4 * 3;
     // Limit decoding attempts to within the maximum length of the known input.
     let limits = Limits::len(decoded_max_len);
+    let mut cursor = Limited::new(Cursor::new(xdr_base64.as_bytes()), limits);
 
-    let value =
-        Type::from_xdr_base64(type_variant, xdr_base64, limits).map_err(|e| format!("{e}"))?;
+    let value = Type::read_xdr_base64_iter(type_variant, &mut cursor)
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(|e| format!("{e}"))?;
+    if value.len() == 1 {
+        let json = serde_json::to_string(&value[0]).map_err(|e| format!("{e}"))?;
+        return Ok(json);
+    }
```

Comment:
This change results in this function returning either a single value, or an array. Where that single value is expected to be an array, the caller also has to check if it's an array, or an array of arrays. The muddying of the return value could be a significant footgun, and it's easier to understand the API if the API returns the same type every time. This is especially the case in the JS environment where there is no typing so it may be easy to code to just one of the return types and not the other.

Could the new functionality be added to a new function, such as `decode_stream`?